src/base/logging.hh

```diff
@@ -123,19 +126,41 @@ class Logger
      */
     [[noreturn]] void exit_helper() { exit(); ::abort(); }
 
+    void
+    registerExtraLog(std::function<std::string()> cb)
+    {
+        extraLog.emplace_back(std::move(cb));
+    }
+
   protected:
     bool enabled;
 
+    std::string
+    getFormattedExtraLog()
+    {
+        std::stringstream ss;
+        for (auto &cb : extraLog) {
+            std::string str = cb();
+            ss << str;
+            if (str.length() && str.back() != '\n' && str.back() != '\r')
+                ss << std::endl;
+        }
+        return ss.str();
+    }
+
     /** Generates the log message. By default it is sent to cerr. */
     virtual void
     log(const Loc &loc, std::string s)
     {
-        std::cerr << loc.file << ":" << loc.line << ": " << s;
+        std::cerr << loc.file << ":" << loc.line << ": " << s
+                  << getFormattedExtraLog();
     }
 
     virtual void exit() { /* Fall through to the abort in exit_helper. */ }
 
     const char *prefix;
+
+    std::vector<std::function<std::string()>> extraLog;
```

Comment:
Could you add doxygen documentation for this variable?

Reply:
Sure. documentation added!